task/backend/analytical_storage.go

```diff
@@ -29,6 +30,79 @@ const (
 	statusTag = "status"
 )
 
+func CreateDestination(log *zap.Logger, cli influxdb.Client, dest DataDestination) error {
+	const numTries = 5
+	for try := 0; ;try++ {
+		// poll for influxdb to come up - usually it should be already up by the time we get here.
+		if try > 0 {
+			time.Sleep(1 * time.Second)
+		}
+		queryForError := func(q string) error {
+			// This query is only successful if the bucket exists
+			result, err := cli.QueryFlux(influxdb.FluxQuery{
+				Org:   dest.Org,
+				OrgID: dest.OrgID,
+				Query: q,
+			})
+			if err != nil {
+				return err
+			}
+			for result.More() {
+				if err := result.Next().Tables().Do(func(table flux.Table) error {
+					return table.Do(func(col flux.ColReader) error {
+						return nil
+					})
+				}); err != nil {
+					return err
+				}
+			}
+			return result.Err()
+		}
+
+		// We do this instead of interpreting the response from `buckets()` because
+		// for 1.x `from(bucket: "mydb")` is valid and means to use the default retention
+		// policy, but `buckets()` will return the fully qualified retention policy.
+		err := queryForError(fmt.Sprintf(`from(bucket: %q) |> range(start: -1s) |> filter(fn: (r) => r._measurement == %q)`, dest.Bucket, dest.Measurement))
+		if err == nil {
+			log.Info("Successfully read from bucket for task analytic store")
+			return nil
+		}
+		log.Info("Error reading from bucket for task analytic storage, attempting to create it", zap.Error(err), zap.String("name", dest.Bucket))
+
+		if try > numTries {
+			break
+		}
+
+		// First check if we're reading any flux queries at all
+		err = queryForError("buckets()")
+		if err != nil {
+			log.Warn("InfluxDB instance for analytic store not processing flux queries", zap.Error(err))
+			continue
+		}
```

Comment:
From the comment, my impression is that if the query fails that means flux is not enabled, so we don't need to create the task run bucket. However, since we `continue` instead of `break`ing or `returning`ing, we'll do this check 4 more times, and increase the startup time by 4 seconds or so.

I'm guessing there could be some errors, like a timeout or connection refused, that warrant retrying this. Are there other errors that allow us to escape from the loop early?

Reply:
If this check fails permanently, we fail kapacitor startup (we already fail kapacitor startup if the influxdb instance is not responding), so I think a bit of extra waiting before failing is ok.

Reply:
In that case, my interpretation is that InfluxDB is up and running, and this check tells us Flux is not enabled, so we should not attempt to create the default database. If that's true, wouldn't `return nil` be appropriate here to speed up starting Kapacitor when Flux is disabled on the InfluxDB instance? Or am I misinterpreting something? Would we never even call this function if Flux is disabled?

Reply:
If you've told kapacitor to store its task logs in InfluxDB, flux needs to be enabled. Configuring Kapacitor to store task logs to an InfluxDB with flux disabled is a serious misconfiguration and we should refuse to start I think - certainly the task logs will not save correctly.

Reply:
In that case, should we return a more helpful `Flux must be enabled` error instead of `could not create database for analytic store`?

Reply:
Both errors will just be logged (since the error will be returned to the startup code) - I think `log.Warn("InfluxDB instance for analytic store not processing flux queries", zap.Error(err))` is sufficient.

Reply:
This looks like:

```
ts=2021-09-16T14:47:54.666-04:00 lvl=info msg="Error reading from bucket for task analytic storage, attempting to create it" service=fluxtask-analytical-store name=kapacitor_fluxtask_logs error="Flux query service disabled. Verify flux-enabled=true in the [http] section o
f the InfluxDB config." errorVerbose="Flux query service disabled. Verify flux-enabled=true in the [http] section of the InfluxDB config.\ngithub.com/influxdata/kapacitor/influxdb.(*HTTPClient).doHttpV2\n\t/Users/sarnold/influx/kapacitor/influxdb/client.go:495\ngithub.com
/influxdata/kapacitor/influxdb.(*HTTPClient).QueryFlux\n\t/Users/sarnold/influx/kapacitor/influxdb/client.go:731\ngithub.com/influxdata/kapacitor/task/backend.CreateDestination.func1\n\t/Users/sarnold/influx/kapacitor/task/backend/analytical_storage.go:42\ngithub.com/infl
uxdata/kapacitor/task/backend.CreateDestination\n\t/Users/sarnold/influx/kapacitor/task/backend/analytical_storage.go:65\ngithub.com/influxdata/kapacitor/task/backend.NewAnalyticalStorage\n\t/Users/sarnold/influx/kapacitor/task/backend/analytical_storage.go:141\ngithub.co
m/influxdata/kapacitor/services/fluxtask.(*Service).Open\n\t/Users/sarnold/influx/kapacitor/services/fluxtask/service.go:71\ngithub.com/influxdata/kapacitor/server.(*Server).startServices\n\t/Users/sarnold/influx/kapacitor/server/server.go:1143\ngithub.com/influxdata/kapa
citor/server.(*Server).Open\n\t/Users/sarnold/influx/kapacitor/server/server.go:1125\ngithub.com/influxdata/kapacitor/cmd/kapacitord/run.(*Command).Run\n\t/Users/sarnold/influx/kapacitor/cmd/kapacitord/run/command.go:156\nmain.(*Main).Run\n\t/Users/sarnold/influx/kapacito
r/cmd/kapacitord/main.go:82\nmain.main\n\t/Users/sarnold/influx/kapacitor/cmd/kapacitord/main.go:42\nruntime.main\n\t/Users/sarnold/.gvm/gos/go1.17/src/runtime/proc.go:255\nruntime.goexit\n\t/Users/sarnold/.gvm/gos/go1.17/src/runtime/asm_amd64.s:1581"
...
ts=2021-09-16T14:47:54.670-04:00 lvl=error msg="encountered error" service=run err="open server: open service *fluxtask.Service: creating analytical store: could not create database for analytic store, tried 5 times"
```

Reply:
I'll accept this